src/app/query-home/search-area/pins/pins.tsx

```diff
@@ -29,11 +28,6 @@ function renderPin(pin: QueryPin, index: number) {
 
 export function Pins() {
   const pins = useSelector(Editor.getPins)
-  const dispatch = useDispatch()
```

Comment:
Since `submitSearch()` now creates a new draft for the current query, this created an infinite loop. Can also fix it by comparing the editor pins to the query, but I thought executing a query after editing a pin didn't match with how we don't execute after editing the value so I removed. Happy to discuss more!

Reply:
Let's discuss! I think we should try to submit searches after every keystroke with a throttle!